index.bs

```diff
@@ -153,6 +156,24 @@ Note: These steps have to be run on the [=file system queue=].
 
 </div>
 
+<div algorithm>
+To <dfn for="file system entry/lock">check for a descendant lock</dfn> on a given
+[=/file system entry=] |entry|:
+
+1. Let |lock| be the |entry|'s [=file system entry/lock=].
+1. If |lock| is not "`open`":
+   1. Return true.
```

Comment:
Currently, "check for a descendant lock" also checks whether _this_ entry is locked. IIUC this would make the "take" algorithm return early if attempting to take a "`shared`" lock on a directory entry, which doesn't seem right?

Reply:
My point is that an algorithm called "check for a descendant lock" shouldn't care whether or not you should be able to take a shared lock on a directory. The fact that it does is a big gotcha IMHO

Reply:
Ah I see good point!